src/lib/math/numbertheory/mod_inv.cpp

```diff
@@ -277,4 +277,117 @@ BigInt inverse_mod(const BigInt& n, const BigInt& mod) {
    return h;
 }
 
+BigInt inverse_modulo_secret_prime(const BigInt& x, const BigInt& p) {
+   BOTAN_ARG_CHECK(x.is_positive() && p.is_positive(), "Parameters must be positive");
+   BOTAN_ARG_CHECK(x < p, "x must be less than p");
+   BOTAN_ARG_CHECK(p.is_odd() and p > 1, "Primes are odd integers greater than 1");
+
+   // TODO possibly use FLT, or the algorithm presented for this case in
+   // Handbook of Elliptic and Hyperelliptic Curve Cryptography
+
+   return inverse_mod_odd_modulus(x, p);
+}
+
+BigInt inverse_modulo_public_prime(const BigInt& x, const BigInt& p) {
+   return inverse_modulo_secret_prime(x, p);
+}
+
+BigInt inverse_modulo_rsa_public_modulus(const BigInt& x, const BigInt& n) {
+   BOTAN_ARG_CHECK(n.is_positive() && n.is_odd(), "RSA public modulus must be odd and positive");
+   BOTAN_ARG_CHECK(x.is_positive() and x < n, "Input must be positive and less than RSA modulus");
```

Comment:
```suggestion
   BOTAN_ARG_CHECK(x.is_positive() && x < n, "Input must be positive and less than RSA modulus");
```

Didn't even know this worked in the general case.

Reply:
Just you wait I'm going to start defining move constructors as `MyClass(MyClass and other)` :imp: 

j/k better to be consistent, I didn't do this intentionally, just a mental autocomplete with Python on my mind I guess.

Reply:
# :fearful: